src/testdir/test_vim9_class.vim

```diff
@@ -3093,25 +3093,78 @@ def Test_closure_in_class()
   v9.CheckSourceSuccess(lines)
 enddef
 
-def Test_call_constructor_from_legacy()
+def Test_construct_object_from_legacy()
+  # Cannot directly invoke constructor from legacy
   var lines =<< trim END
     vim9script
 
     var newCalled = 'false'
 
     class A
-      def new()
+      def new(arg: string)
         newCalled = 'true'
       enddef
     endclass
 
-    export def F(options = {}): any
-      return A
+    export def CreateA(...args: list<any>): A
+      return call(A.new, args)
     enddef
 
-    g:p = F()
-    legacy call p.new()
+    g:P = CreateA
+    legacy call g:P('some_arg')
     assert_equal('true', newCalled)
+    unlet g:P
+  END
+  v9.CheckSourceSuccess(lines)
+
+  lines =<< trim END
+    vim9script
+
+    var newCalled = 'false'
```

Comment:
Can you use a Boolean variable instead of a string true/false?

Reply:
OK; makes sense; DONE. Also changed the original test that I cloned from.